octobot_evaluators/evaluator/abstract_evaluator.py

```diff
@@ -105,37 +115,61 @@ def get_is_time_frame_wildcard(cls) -> bool:
+        available_time_frames = [self.time_frame]
+        if self.get_is_cryptocurrencies_wildcard():
+            currencies = all_symbols_by_crypto_currencies.keys()
+        if self.get_is_symbol_wildcard():
+            symbols = [currency_symbol
+                       for currency_symbols in all_symbols_by_crypto_currencies.values()
+                       for currency_symbol in currency_symbols]
+        if self.get_is_time_frame_wildcard():
+            available_time_frames = time_frames
+        return currencies, symbols, available_time_frames
+
+    def initialize(self, all_symbols_by_crypto_currencies, time_frames, real_time_time_frames):
+        currencies, symbols, time_frames = self._get_tentacle_registration_topic(all_symbols_by_crypto_currencies,
+                                                                                 time_frames,
+                                                                                 real_time_time_frames)
+        for currency in currencies:
+            for symbol in symbols:
+                if symbol is None or symbol in all_symbols_by_crypto_currencies[currency]:
+                    for time_frame in time_frames:
+                        set_tentacle_value(
+                            matrix_id=self.matrix_id,
+                            tentacle_type=self.get_eval_type(),
+                            tentacle_value=None,
+                            tentacle_path=get_matrix_default_value_path(
+                                exchange_name=self.exchange_name,
+                                tentacle_type=self.evaluator_type.value,
+                                tentacle_name=self.get_name(),
+                                cryptocurrency=currency,
+                                symbol=symbol,
+                                time_frame=time_frame.value if time_frame else None
+                            )
+                        )
 
     async def evaluation_completed(self,
                                    cryptocurrency: str = None,
                                    symbol: str = None,
                                    time_frame=None,
-                                   eval_note=None) -> None:
+                                   eval_note=None,
+                                   eval_time=0,
+                                   notify=True) -> None:
         """
         Main async method to notify matrix to update
         :param cryptocurrency: evaluated cryptocurrency
         :param symbol: evaluated symbol
-        :param time_frame: evaluated time frame
+        :param time_frame: evaluated time fram
         :param eval_note: if None = self.eval_note
+        :param eval_time: the time of the evaluation if relevant, default is 0
         :return: None
         """
         try:
-            if not eval_note:
-                eval_note = self.eval_note if self.eval_note else START_PENDING_EVAL_NOTE
+            if eval_note is None:
```

Comment:
Why not keeping `not eval_note` ? It's larger than `eval_note is None`

Reply:
Right but it prevents boolean type eval notes

Reply:
Right ! :+1: